### ISSUE TITLE
Improvements to the base entry fee

### ISSUE THREAD

**Issue by jfly** [thewca/worldcubeassociation.org#3105]
Currently, we have the following ui on the competition form:

![image](https://user-images.githubusercontent.com/277474/42831024-155c0ce4-89a2-11e8-9a94-4c27a764bb0a.png)

There are a couple of issues with this form:

1. We have a database default that sets this currency code to USD. That's not very worldly of us, although maybe it's a pretty reasonable to default and in practice saves people clicking.
2. @AlbertoPdRF recently added explicit support for free competitions, which is great! However, it's kind of weird that you have to select a currency when marking a competition as free. $0 USD is the same thing as $0 Canadian, which is the same thing as 0 Bitcoin, ... A better UI might be to have an explicit "this competition is free" checkbox or dropdown, and only if the competition is *not* free, do we ask for a currency and an amount.
3. Being forced to choose a currency is also a bit weird for old competitions that don't have a base entry fee set. I think it's less important to improve this UX, however.

To address 1) and 3), I think it would be better for us to handle currency in the same way we handle all other competition attributes: it starts out NULL, but is required once you confirm the competition.

**Comment by dunkOnIT** [comment]
@gregorbg I'm assuming this is still relevant to us despite the work we've done with payments?

**Comment by gregorbg** [comment]
Yes, although it has nothing to do with payments (and payment services) in the stricter sense. This issue is only concerned with how to let organizers/Delegates input base fees, which is kinda the prerequisite to payments
